superset/viz.py

```diff
@@ -771,22 +783,29 @@ def get_data(self, df: pd.DataFrame) -> VizData:
         if self.form_data.get("granularity") == "all" and DTTM_ALIAS in df:
             del df[DTTM_ALIAS]
 
-        aggfunc = self.form_data.get("pandas_aggfunc") or "sum"
-
-        # Ensure that Pandas's sum function mimics that of SQL.
-        if aggfunc == "sum":
-            aggfunc = lambda x: x.sum(min_count=1)
+        metrics = [utils.get_metric_name(m) for m in self.form_data["metrics"]]
+        aggfuncs: Dict[str, Union[str, Callable]] = {}
+        for metric in metrics:
```

Comment:
Would be nice to break this out into a functional method and adding unit tests for it